sophia/src/term/_display.rs

```diff
@@ -29,12 +29,16 @@ where
             write!(w, "{}", iri)?;
             w.write_char('>')?;
         }
-        BNode(ident) => {
-            w.write_str("_:")?;
-            if ident.is_n3() {
-                w.write_str(ident.as_ref())?;
+        BNode(bn) => {
+            if bn.is_n3() {
+                bn.write_fmt(w)?;
             } else {
-                write_non_n3_bnode_id(w, ident.as_ref())?;
+                // non conformant identifier
+                w.write_str("_:_")?;
+                for b in bn.as_ref().as_bytes() {
+                    write!(w, "{:x}", b)?;
+                }
+                w.write_str("_:_")?;
```

Comment:
Is this still necessary? With this PR it is already checked at construction if a blank node is valid according to N3, therefore, there should no longer be non-conformant blank nodes.

Reply:
Agreed. At some point I wanted to be more lenient with bnode identifiers, but that adds complexity, and is inconsistent with how we handle variables.